patchset/0030-lavc-vaapi_hevc-fill-rext-luma-chroma-offset-in-the-.patch

```diff
@@ -0,0 +1,107 @@
+--- a/libavcodec/vaapi_hevc.c
++++ b/libavcodec/vaapi_hevc.c
+@@ -322,10 +322,17 @@ fail:
+     return ret;
+ }
+ 
+-static void fill_pred_weight_table(const HEVCContext *h,
++static void fill_pred_weight_table(const AVCodecContext *avctx,
++                                   const HEVCContext *h,
+                                    const SliceHeader *sh,
+                                    VASliceParameterBufferHEVC *slice_param)
+ {
++#if VA_CHECK_VERSION(1, 2, 0)
++    int is_rext = avctx->profile == FF_PROFILE_HEVC_REXT;
++#else
++    int is_rext = 0;
++#endif
++
+     int i;
+ 
+     memset(slice_param->delta_luma_weight_l0,   0, sizeof(slice_param->delta_luma_weight_l0));
+@@ -353,21 +360,26 @@ static void fill_pred_weight_table(const HEVCContext *h,
+ 
+     for (i = 0; i < 15 && i < sh->nb_refs[L0]; i++) {
+         slice_param->delta_luma_weight_l0[i] = sh->luma_weight_l0[i] - (1 << sh->luma_log2_weight_denom);
+-        slice_param->luma_offset_l0[i] = sh->luma_offset_l0[i];
++
+         slice_param->delta_chroma_weight_l0[i][0] = sh->chroma_weight_l0[i][0] - (1 << sh->chroma_log2_weight_denom);
+         slice_param->delta_chroma_weight_l0[i][1] = sh->chroma_weight_l0[i][1] - (1 << sh->chroma_log2_weight_denom);
+-        slice_param->ChromaOffsetL0[i][0] = sh->chroma_offset_l0[i][0];
+-        slice_param->ChromaOffsetL0[i][1] = sh->chroma_offset_l0[i][1];
++        if (!is_rext) {
++            slice_param->luma_offset_l0[i] = sh->luma_offset_l0[i];
++            slice_param->ChromaOffsetL0[i][0] = sh->chroma_offset_l0[i][0];
++            slice_param->ChromaOffsetL0[i][1] = sh->chroma_offset_l0[i][1];
++        }
+     }
+ 
+     if (sh->slice_type == HEVC_SLICE_B) {
+         for (i = 0; i < 15 && i < sh->nb_refs[L1]; i++) {
+             slice_param->delta_luma_weight_l1[i] = sh->luma_weight_l1[i] - (1 << sh->luma_log2_weight_denom);
+-            slice_param->luma_offset_l1[i] = sh->luma_offset_l1[i];
+             slice_param->delta_chroma_weight_l1[i][0] = sh->chroma_weight_l1[i][0] - (1 << sh->chroma_log2_weight_denom);
+             slice_param->delta_chroma_weight_l1[i][1] = sh->chroma_weight_l1[i][1] - (1 << sh->chroma_log2_weight_denom);
+-            slice_param->ChromaOffsetL1[i][0] = sh->chroma_offset_l1[i][0];
+-            slice_param->ChromaOffsetL1[i][1] = sh->chroma_offset_l1[i][1];
++            if (!is_rext) {
++                slice_param->luma_offset_l1[i] = sh->luma_offset_l1[i];
++                slice_param->ChromaOffsetL1[i][0] = sh->chroma_offset_l1[i][0];
++                slice_param->ChromaOffsetL1[i][1] = sh->chroma_offset_l1[i][1];
++            }
+         }
+     }
+ }
+@@ -462,7 +474,7 @@ static int vaapi_hevc_decode_slice(AVCodecContext *avctx,
+             last_slice_param->RefPicList[list_idx][i] = get_ref_pic_index(h, rpl->ref[i]);
+     }
+ 
+-    fill_pred_weight_table(h, sh, last_slice_param);
++    fill_pred_weight_table(avctx, h, sh, last_slice_param);
```

Comment:
Is it necessary to modify this function for Non-rext profile?
Filling the luma_offset[] and chromaOffset[] for both last_slice_param.base and last_slice_param.rext seems to introduce less changing. (It's true that base.luma/chroma offset would be redundant for rext, however the effort seems to be acceptable)

Reply:
It will introduce junk information in VASliceParameterBufferHEVC, a reader not know details about it will confuse, why we need to fill the same things twice?